certbot-zerossl.sh

```diff
@@ -7,7 +7,14 @@ function parse_eab_credentials()
     PYTHONIOENCODING=utf8
     ZEROSSL_EAB_KID=$(echo $1 | python -c "import sys, json; print(json.load(sys.stdin)['eab_kid'])")
     ZEROSSL_EAB_HMAC_KEY=$(echo $1 | python -c "import sys, json; print(json.load(sys.stdin)['eab_hmac_key'])")
-    CERTBOT_ARGS+=(--eab-kid "$ZEROSSL_EAB_KID" --eab-hmac-key "$ZEROSSL_EAB_HMAC_KEY" --server "https://acme.zerossl.com/v2/DV90")
+    
+    # Check if hmac key starts with dash - this will confuse python or certbot, see https://github.com/certbot/certbot/issues/7114
+	if  [[ $ZEROSSL_EAB_HMAC_KEY != -* ]];	then
+	    GOOD_KEY_FOUND=true
+	    CERTBOT_ARGS+=(--eab-kid "$ZEROSSL_EAB_KID" --eab-hmac-key "$ZEROSSL_EAB_HMAC_KEY" --server "https://acme.zerossl.com/v2/DV90")
+    else
+        echo "Got hmac key with leading dash, retrying..."
```

Comment:
Technically that dash is a hyphen. ;)

https://datatracker.ietf.org/doc/html/rfc20#section-4.2
https://owl.purdue.edu/owl/english_as_a_second_language/esl_students/punctuation/hyphens_and_dashes.html